third_party/proton/csrc/include/Driver/GPU/CudaApi.h

```diff
@@ -21,6 +21,8 @@ template <bool CheckSuccess> CUresult deviceGet(CUdevice *device, int ordinal);
 
 Device getDevice(uint64_t index);
 
+const std::string getCudaArchName(int arch);
```

Comment:
We probably don't need this function.

We could convert the type of `arch` from `uint64_t` to `std::string`.

Then for CUDA devices, we do:

- `CUDA_80`, `CUDA_90`, ....

For hip devices, we do:

- `HIP_gfx90a`, `HIP_gfx941`, ...

Reply:
Otherwise, for now, we have a field `arch` not used for HIP devices.

Reply:
So you're thinking get rid of "arch" in the data map entirely and just replace it with the "arch_name" string for both Cuda and HIP?

Reply:
Yes

Reply:
I think even the device type prefix is not necessary because we already have the field.

So we can just do, "80", "90", ... for CUDA and "gfx90a", "gfx941" for HIP

Reply:
Agreed

Reply:
fyi: I was having similar issues before in the main compiler: https://github.com/triton-lang/triton/pull/3741. End up choosing `<backend>:<arch>`, for example, `cuda:80`, `hip:gfx942`.

Reply:
OK. Similar spirt here.